### PR TITLE
Rewrite data fetching from scratch

### PR DESCRIPTION
**Pull request by jhorbowicz**
Because the API changed a lot since this version
was written, I had to rewrite most of the
data fetching part. I also updated few types.
Usually I would separate it into a lot of smaller
commits, but it was hard to do in the manner, that every
commit could work separately, so I just created big
one with all the cnahges.

See #2 for additional info

### CODE REVIEW

File: pages/apps/details/[appDetails].tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  return (
+    <Main>
+      <ApplicationDetails data={data} />
+    </Main>
+  )
+}
+
+export const getStaticProps: GetStaticProps = async ({params: {appDetails}}) => {
+  console.log("Fetching data for app details: ", appDetails)
+  const appData = await fetch(`https://flathub-backend.openshift.gnome.org/master/appstream/${appDetails}`)
```
Comment (bilelmoussaoui) [inline pages/apps/details/[appDetails].tsx:15]: I would avoid hardcoding the server URL here

**Comment by bilelmoussaoui** [comment]
Thanks a lot for the PR! It's fine as a single commit for now, we are still in the early stages of the rewrite I would say. I will review it in details tomorrow.

File: pages/apps/details/[appDetails].tsx (hunk 1 above)
Comment (jhorbowicz) [reply pages/apps/details/[appDetails].tsx:15]: Sure, nice catch, I must have missed it :)

**Comment by jhorbowicz** [comment]
@bilelmoussaoui, cool, I will try to clean it a little tomorrow, I just wanted to drive it to the point I would be able to build it or run it in dev mode. There are also few console.logs, some of them useful, but some of them could be removed once we will ensure everything works and is reproducible.

File: pages/index.tsx [hunk 2]
```diff
@@ -45,12 +52,17 @@ export default function Home({ recentlyUpdated }) {
 }
 
 export const getStaticProps: GetStaticProps = async () => {
-  const res = await fetch(`${BASE_URI}/apps/collection/recently-updated/7`)
-  const recentlyUpdated: Application[] = await res.json()
+  const recentlyUpdated = await fetchCollection(Collections.recenltyUpdated, 6)
```
Comment (bilelmoussaoui) [inline pages/index.tsx:55]: Maybe put `6` in a const, as we re-use that value in multiple places

File: src/fetchers.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import Appstream from './types/Appstream'
+import Collections, {Collection} from './types/Collection'
+import Category from './types/Category'
+
+import {POPULAR_URL, APP_DETAILS, RECENTLY_UPDATED_URL, EDITORS_PICKS_APPS_URL, EDITORS_PICKS_GAMES_URL, APPSTREAM_URL, CATEGORY_URL, SEARCH_APP} from './env'
+
+const fetchEntry = async (entry: string): Promise<Appstream | {}> => {
+  let entryJson: Appstream | {};
```
Comment (bilelmoussaoui) [inline src/fetchers.ts:8]: There's a weird combination of either using `;` or not, we should setup a linter and keep the code consistent on that regard

**Review by bilelmoussaoui — COMMENTED** [review]
Otherwise looks good to me, thanks!

File: src/fetchers.ts (hunk 3 above)
Comment (jhorbowicz) [reply src/fetchers.ts:8]: Yeah, unfortunately I have a habit of putting semicolon by myself, but some linter and maybe prettier is a cool idea, will create issue for that.

Comment (bilelmoussaoui) [reply src/fetchers.ts]: Aha, I tend to use the standardjs linter which doesn't use the semicolon :-) but I don't mind that much, it's just details. We  can set that at another time.

**Comment by bilelmoussaoui** [comment]
I will go ahead and merge this one. I will probably not have the time to work on anything myself the next couple of weeks, so if there are anything that might interests, please go ahead and work on it. I will try to review/merge as soon as possible. 

Thank a lot for the patch!

**Pull request merged**